src/Thread/BoundedBlockingBuffer.java

```diff
@@ -0,0 +1,44 @@
+package Thread;
+
+public class BoundedBlockingBuffer<T> {
+    T data;
+    boolean isEmpty = true;
+
+    public synchronized T take() {
+        while (isEmpty) {
+            if (this.data == null) {
+                try {
+                    wait();
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            } else {
+                isEmpty = false;
+                this.data = data;
+                System.out.println("Take " + this.data);
+                notify();
+            }
+
+        }
+        return this.data;
+    }
+
+
+    public synchronized void put(T data) {
+        while (isEmpty) {
```

Comment:
Put method, you can write like this:

 while (isEmpty) {
            try {
                wait();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
        isEmpty = true;
        this.data = data;
        System.out.println("Put " + this.data);
        notify();
    }